packages/msw/src/mswPlugin.ts

```diff
@@ -0,0 +1,32 @@
+import { Plugin, type AbstractRuntime } from "@squide/core";
+import type { RestHandler } from "msw";
+import { RequestHandlerRegistry } from "./requestHandlerRegistry.ts";
+
+export class MswPlugin extends Plugin {
+    readonly #requestHandlerRegistry = new RequestHandlerRegistry();
+    #isMswStarted = false;
+
+    constructor() {
+        super(MswPlugin.name);
+    }
+
+    registerRequestHandlers(handlers: RestHandler[]) {
+        this.#requestHandlerRegistry.add(handlers);
+    }
+
+    get requestHandlers(): RestHandler[] {
+        return this.#requestHandlerRegistry.handlers;
+    }
+
+    setAsStarted() {
```

Comment:
I would expect the method to be called `start()`. `setAsStarted `sounds weird

Reply:
I think it might be confusing to call it "start" because it is not starting anything, nor executing anything. It's a state transition to switch a shared variable from "pending" to "started".

Reply:
When you start a plugin, it's state transition from "pending" to "started", it doesn't have to do anything. I agree with Alex here

Reply:
Sorry, I still don't get it, there's no notion of starting a plugin, it's about the status of the Mock service worker.

Reply:
i mean i understand what you are saying, but at the same time you have a function right under that is "isStarted" a plugin can not be started so why is there an isStarted function? 

Maybe it's the entire variable that doesn't make sense to keep here. Why is this class responsible of keeping the started state of msw

Reply:
Because the `useIsMswStarted` hook needs a way to know when MSW has been started by the host application bootstrapping code so it can update is state and force a re-render of the application.

That's similar to the `useAreModulesReady` hook.

I thought that doing this through the plugin would be a quick & easy way to handle this. An alternative would be to instead export a `setMswAsStarted` function from the `useIsMswStarted` hook file.

Reply:
I updated the code with what I suggested in my previous message ☝🏻 . That already seems way better, thank you 🙏🏻